test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthCompilationTests.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Dnx.CommonTestUtils;
+using Microsoft.Dnx.DesignTimeHost.FunctionalTests.Infrastructure;
+using Microsoft.Dnx.Runtime;
+using Microsoft.Dnx.Tooling;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Dnx.DesignTimeHost.FunctionalTests
+{
+    [Collection(nameof(DthFunctionalTestCollection))]
+    public class DthCompilationTests
+    {
+        private readonly DthFunctionalTestFixture _fixture;
+
+        public DthCompilationTests(DthFunctionalTestFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        public static IEnumerable<object[]> RuntimeComponents
+        {
+            get
+            {
+                foreach (var runtime in TestUtils.GetClrRuntimeComponents())
+                {
+                    yield return runtime;
+                }
+
+                if (!RuntimeEnvironmentHelper.IsMono)
+                {
+                    foreach (var runtime in TestUtils.GetCoreClrRuntimeComponents())
+                    {
+                        yield return runtime;
+                    }
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(RuntimeComponents))]
+        public void DthCompilation_GetDiagnostics_OnEmptyConsoleApp(string flavor, string os, string architecture)
+        {
+            var projectName = "EmptyConsoleApp";
+            var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+            var testProject = _fixture.GetTestProjectPath(projectName);
+
+            using (var server = DthTestServer.Create(runtimeHomePath, testProject))
+            using (var client = new DthTestClient(server, 0))
+            {
+                // Drain the inital messages
+                client.Initialize(testProject);
+                client.DrainMessages();
+
+                client.SendPayLoad("GetDiagnostics", new { });
+                var message = client.DrainMessages().Single(msg => msg.MessageType == "AllDiagnostics");
+                message.EnsureSource(server, client);
+
+                var diagnosticsGroups = (JArray)message.Payload;
+                Assert.Equal(4, diagnosticsGroups.Count);
+
+                foreach (var df in diagnosticsGroups.OfType<JObject>())
+                {
+                    if (RuntimeEnvironmentHelper.IsMono && df["Framework"]["ShortName"].Value<string>() == "net46")
```

Comment:
Because other frameworks build on Mono. I can drop the `net46` framework in the sample.